src/init.rs

```diff
@@ -158,6 +158,12 @@ if [[ ${preexec_functions[(ie)starship_preexec]} -gt ${#preexec_functions} ]]; t
     preexec_functions+=(starship_preexec);
 fi;
 STARSHIP_START_TIME="$(date +%s)";
+function zle-keymap-select
+{
+    PROMPT=$(starship prompt --keymap=$KEYMAP)
+    zle reset-prompt
+}
+zle -N zle-keymap-select
```

Comment:
Unfortunately, this is going to crash `zsh` when you try to eval, because the *entire* script is evaluated on one line.

There is a PR in progress to fix that, but until that's merged, we'll need to spam semicolons everywhere for this to successfully `eval`.

```
function zle-keymap-select
{
    PROMPT=$(starship prompt --keymap=$KEYMAP);
    zle reset-prompt;
};
zle -N zle-keymap-select;
```